File: Microsoft-prototype/ToolWithSizes.cpp
#include "ToolWithSizes.h" 
#include "StringData.h"
ToolWithSizes::ToolWithSizes(const std::vector<string>& toolNames, const ToolHandleType& toolHandle, const std::vector<Length>& toolLengths) : Tool(toolNames, toolHandle), toolLengths_(toolLengths) {
}

void ToolWithSizes::searchProperty(const string& speechTranscription, ToolPropertySearch& toolPropertySearch)const {
	const bool nameFound = Tool::searchName(speechTranscription, toolPropertySearch);
	const bool oldNameFound = Tool::searchOldName(toolPropertySearch);
	if (nameFound) {
		searchLength(speechTranscription, toolPropertySearch);
		toolPropertySearch.isToolFound = true;
	}
	if (oldNameFound) {
		searchLength(speechTranscription, toolPropertySearch);
	}
}

void ToolWithSizes::searchLength(const string& speechTranscription, ToolPropertySearch& toolPropertySearch)const {
	for (const auto& length : toolLengths_) {
		if (speechTranscription.find(lengthToString[length]) != string::npos) {
			toolPropertySearch.length = length;
			toolPropertySearch.toolNeedSize = "";
			toolPropertySearch.isToolFound = true;
			return;
		}
	}
	toolPropertySearch.toolNeedSize = this->toolNames_[0];
}

File: vosk-prototype/ToolWithSizes.py
from Tool import Tool


class ToolWithSizes(Tool):
    def __init__(self, tool_names, tool_handle, toollengths):
        self.toollengths_ = toollengths
        super().__init__(tool_names, tool_handle)

    def search_property(self, speech_transcription, tool_property_search):
        namefound = super(ToolWithSizes, self).searchname(speech_transcription, tool_property_search)
        oldnamefoud = super(ToolWithSizes, self).searcholdname(tool_property_search)
        if namefound:
            self.searchlength(speech_transcription, tool_property_search)
            tool_property_search.is_tool_found_ = True
        if oldnamefoud:
            self.searchlength(speech_transcription, tool_property_search)

    def searchlength(self, speechTranscription, toolPropertySearch):
        for length in self.toollengths_:
            if speechTranscription.find(length.value) != -1:
                toolPropertySearch.length_ = length
                toolPropertySearch.toolneedsize_ = ""
                toolPropertySearch.is_tool_found_ = True
                return
        toolPropertySearch.toolneedsize_ = self.tool_names[0]
